overlord/configstate/configcore/runwithstate.go

```diff
@@ -64,6 +64,9 @@ func init() {
 
 	// netplan.*
 	addWithStateHandler(validateNetplanSettings, handleNetplanConfiguration, &flags{coreOnlyConfig: true})
+
+	// kernel.{,dangerous-}cmdline-append
+	addWithStateHandler(validateCmdlineAppend, handleCmdlineAppend, coreOnly)
```

Comment:
actually for our use case we need to support this also with classic+modes, we might need to expand flags for that?

Reply:
Right, that is definitely needed. I've added a TODO as the PR is big enough, I can tackle this in another PR.